### PR TITLE
Feature/732 Datepicker

### PR DESCRIPTION
**Pull request by LeitnerMi**
**Prerequisites: Always create two feature branches (frontend + backend) even if your feature only affects one of them**

**DEV:**
> Development
- [ ] Acceptance tests by dev [OpenShift feature environment]
  - [ ] Moved to "Ready for review"

> Review
- [ ] Pull-request reviewed + approved (including Code Coverage) by reviewer [GitHub]
- [ ] Inform dev after review through MEGA-PR (by reviewer) [Chat Space]
- [ ] Version upgrade by dev [GitHub Issue Milestone + maven revision]
- [ ] Merge to main by dev [GitHub]
  - [ ] Moved to "Ready for deployment"

> Deployment
- [ ] Integration and Unit tests on main environment passed [Workflow]
- [ ] Deployed to the test environment [GitOps Workflow triggered]
  - [ ] Moved to "Ready for testing"

**PO:**
> Testing
- [ ] Acceptance tests on the test enviroment passed
- [ ] Inspected and approved
  - [ ] Moved to "Ready for release"

### CODE REVIEW

File: libs/shared/ui/src/lib/date-picker-custom-header/date-picker-custom-header.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { DatePickerCustomHeaderComponent } from './date-picker-custom-header.component';
+
+describe('DatePickerCustomHeaderComponent', () => {
+  let component: DatePickerCustomHeaderComponent;
```
Comment (Ruhsi) [inline libs/shared/ui/src/lib/date-picker-custom-header/date-picker-custom-header.component.spec.ts]: I think you need the generics here as well:
DatePickerCustomHeaderComponent<any>

File: libs/shared/ui/src/lib/date-picker-custom-header/date-picker-custom-header.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  let fixture: ComponentFixture<DatePickerCustomHeaderComponent>;
```
Comment (Ruhsi) [inline libs/shared/ui/src/lib/date-picker-custom-header/date-picker-custom-header.component.spec.ts]: and here: 
let fixture: ComponentFixture<DatePickerCustomHeaderComponent<any>>

File: libs/shared/ui/src/lib/date-picker-custom-header/date-picker-custom-header.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [DatePickerCustomHeaderComponent],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(DatePickerCustomHeaderComponent);
```
Comment (Ruhsi) [inline libs/shared/ui/src/lib/date-picker-custom-header/date-picker-custom-header.component.spec.ts]: fixture = TestBed.createComponent(DatePickerCustomHeaderComponent<any>);

File: libs/shared/ui/src/lib/datepicker-month-year/datepicker-month-year.component.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    MatDateFnsModule,
+    MatDatepickerModule,
+    ReactiveFormsModule,
+    MatInputModule,
+  ],
+})
+export class DatepickerMonthYearComponent {
+  @Input({ required: true }) date!: Date;
+  @Output() dateChanged: EventEmitter<Date> = new EventEmitter<Date>();
+  dateFormControl = new FormControl(this.date);
```
Comment (Ruhsi) [inline libs/shared/ui/src/lib/datepicker-month-year/datepicker-month-year.component.ts]: don't know the conventions but some with type some without?

**Review by Ruhsi — CHANGES_REQUESTED** [review]

**Comment by Ollitod** [comment]
1. DatePickerCustomHeaderComponent should be moved to the directory of DatepickerMonthYearComponent as a subdirectory.

**Pull request merged**